DANE_s_JSONem/src/engeto/company/com/ListOfStatesTaxes.java

```diff
@@ -0,0 +1,121 @@
+package engeto.company.com;
+
+import org.json.simple.parser.JSONParser;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.json.simple.JSONObject;
+import org.json.simple.parser.ParseException;
+
+public class ListOfStatesTaxes {
+
+    String source;
+    List<State> importedStatesTaxes;
+
+    public ListOfStatesTaxes(String source) {
+        this.source = source;
+        this.importedStatesTaxes = new ArrayList<>();
+    }
+
+    public void setTaxRatesOfCountriesFromJSON() throws IOException {
+        JSONParser jsonParser = new JSONParser();
+
+        try (FileReader reader = new FileReader(source)) {
+
+            // dostaneme JSON do objektu
+            Object javaObject = new JSONParser().parse(reader);
+
+            // cast java objektu do JSONObjektu
+            JSONObject jsonObject = (JSONObject) javaObject;
+
+            // vytáhni jen objekt "rates"
+            Map rates = (Map)jsonObject.get("rates");
+
+            for (Map.Entry state : (Iterable<Map.Entry>) rates.entrySet()) {
+
+                // založ nový stát a naplň atribut pro zkratku
+                State newState = new State();
+                newState.setStateShort(state.getKey().toString());
+
+                // vytáhni pro každý stát jeho atributy
+                Map stateInfo = (Map)rates.get(newState.getStateShort());
+
+                for (Map.Entry stateInfoIt : (Iterable<Map.Entry>) stateInfo.entrySet()){
+
+                    switch(stateInfoIt.getKey().toString()){
+                        case "country":
+                            newState.setStateLong(stateInfoIt.getValue().toString());
+                            break;
+                        case "standard_rate":
+                            newState.setBaseRate(Double.parseDouble(stateInfoIt.getValue().toString()));
+                            break;
+                        case "reduced_rate":
+                            if(stateInfoIt.getValue().toString().equals("false")){
```

Comment:
určitě existuje metoda k získání přímo boolean + místo if použít ternární operátor